File: C_plus_plus/Remove_Element.cpp
class Solution {
public:
    int removeElement(int A[], int n, int elem) {
        int last = n-1;
        for(int i = 0 ; i <= last ; ++i){
            if(A[i] == elem){
                while(A[last]==elem&&last>i) --last;
                A[i] = A[last];
                --last;
            }
        }
        return last+1;
    }
};

File: java/Remove_Element.java
// one pass
public class Solution {
    public int removeElement(int[] nums, int val) {
        int length = 0;
        for (int i = 0 ; i < nums.length; ++i) {
            if (nums[i] == val) continue;
            nums[length++] = nums[i];
        }
        return length;
    }
}
